structuralcodes/sections/specific.py

```diff
@@ -0,0 +1,446 @@
+            │          │           │
+            │          │           │
+            │          │           │
+            │          │           │
+            │          └───────────┼─────►
+            │                      │     y
+            │                      │
+            │                      │
+            │                      │
+            │                      │
+            │                      │
+            │                      │
+            └──────────────────────┘
+
+            RectangularRCSection.fromGeometry(Geometry(itPolygons,itMaterials))
+
+
+    """
+
+    def __init__(
+        self,
+        width: float,
+        height: float,
+        cover: float,
+        reinforcement_data: ReinforcementData,
+        concrete: Concrete,
+        reinforcement: ElasticPlastic,
+        confined_concrete: t.Union[Concrete, str, None] = None,
+        stirrups_data: t.Optional[StirrupsData] = None,
+        name: t.Optional[str] = None,
+    ) -> None:
+        if name is None:
+            name = 'RectangularRCSection'
+        super().__init__(name)
+
+        # section geometry
+        self.height = height
+        self.width = width
+        self.cover = cover
+
+        # materials
+        self.concrete = concrete
+        self.reinforcement = reinforcement
+        if confined_concrete is not None:
+            if isinstance(confined_concrete, str):
+                # provided a string, check if it auto
+                if (
+                    confined_concrete.lower() == 'automatic'
+                    or confined_concrete.lower() == 'auto'
+                ):
+                    # TODO: compute automatic confinement
+                    if stirrups_data is None:
+                        raise ValueError(
+                            'Asked for automatic confinement \
+                            computation, but stirrups_data is None \
+                            \nPlease provide a proper stirrups_data'
+                        )
+                    self.confined_concrete = concrete
+                else:
+                    raise ValueError(f'Value "{confined_concrete}" unknown')
+            elif isinstance(confined_concrete, Concrete):
+                # provided a Concrete material, used that one
+                self.confined_concrete = confined_concrete
+        else:
+            self.confined_concrete = concrete
+
+        # stirrups
+        if stirrups_data is None:
+            # If no stirrups are provided by default it is assumed the
+            # diameter is 8 mm - The spacing is not significant
+            stirrups_data = StirrupsData(8, 150)
+        self.stirrups_data = stirrups_data
+
+        # longitudinal reinforcement
+        self.reinforcement_data = reinforcement_data
+        # arrange in the section the reinforcement
+        self._arrange_reinforcement_in_section()
+        # For now discretize in fibers here
+        self._discretize_fibers(10, 25)
```

Comment:
I don't think we need for now to do discretization. 
We can add it later: we could avoid triangulation in the fiber integrator computing a quad triangulation.